### ISSUE TITLE
Development/Docker version of configuration files

### ISSUE THREAD

**Issue by dannymk** [AlliumCepa/webgui#38]
We want a "development" version of the configuration files so that the docker-compose file will work out of the box.

The docker compose file would then mount these files as follows:
```
...
   allium-cepa:
     image: webgui:local
     container_name: allium
     depends_on:
      - db
     volumes:
      - ./etc/WebGUI.conf.docker:/WebGUI/etc/WebGUI.conf
      - ./etc/log.conf.docker:/WebGUI/etc/log.conf
      - ./etc/spectre.conf.docker:/WebGUI/etc/spectre.conf
      - ./lib:/WebGUI/lib
      - ./www:/WebGUI/www
      # If you are going to make changes and improvements to app.psgi and others you should include those for testing
      #   here as well
      # - ./app.psgi:/WebGUI/app.psgi
      # - ./share/site.psgi:/WebGUI/share/site.psgi      
      # - ./t:/WebGUI/t
     ports:
      - "80:5000"
...
```

**Comment by scrottie** [comment]
Respectfully disagree.  log.conf and spectre.conf shouldn't need to be edited in most cases.  We can package them inside the docker container.  Requiring people to have their own copies on the docker host makes it harder.  Even the mariadb image has its config file inside the docker image.

And same thing for webgui.conf.  We should ship one that "just works" that's already inside the /WebGUI/etc directory.

What we can do and I suggest we do is document how to make etc a volume if people *want* to do something more complex.  If they want to host multiple sites, then they need to create multiple databases and set up users etc for them in the mariadb image or otherwise.  I don't want to try to automate that right now.  We should focus on the admin.

Getting spectre working is a todo item.

Is that plan reasonable?  Ship working default config files and let people turn etc in to a volume and replace them if they want?

**Comment by dannymk** [comment]
The problem with not providing a reasonable default set of files for the docker-compose will rob an unexperienced end user of a quick experience/evaluation. 

We can also do the following if we change some of the values in the original configuration files:

```
   allium-cepa:
     image: webgui:local
     container_name: allium
     depends_on:
      - db
     volumes:
      - ./etc/WebGUI.conf.original:/WebGUI/etc/WebGUI.conf
      - ./etc/log.conf.original:/WebGUI/etc/log.conf
      - ./etc/spectre.conf.original:/WebGUI/etc/spectre.conf
      - ./lib:/WebGUI/lib
      - ./www:/WebGUI/www
      # If you are going to make changes and improvements to app.psgi and others you should include those for testing
      #   here as well
      # - ./app.psgi:/WebGUI/app.psgi
      # - ./share:/WebGUI/share/site.psgi      
      # - ./t:/WebGUI/t
     ports:
      - "80:5000"
```

As it stands right now, unless the user provides valid configuration files the image will fail to run.

**Comment by dannymk** [comment]
This has been resolved in commit:  d1dfcd98fc3f2e377f365bd1c7f1cffa5be887d8
